Brokerages/Oanda/OandaBrokerage.cs

```diff
@@ -0,0 +1,1052 @@
+                    var fill = new OrderEvent(qcOrder, "Oanda Fill Event")
+                    {
+                        Status = OrderStatus.Filled,
+                        //    // this is guaranteed to be wrong in the event we have multiple fills within our polling interval,
+                        //    // we're able to partially cope with the fill quantity by diffing the previous info vs current info
+                        //    // but the fill price will always be the most recent fill, so if we have two fills with 1/10 of a second
+                        //    // we'll get the latter fill price, so for large orders this can lead to inconsistent state
+                        FillPrice = (decimal) data.transaction.price,
+                        FillQuantity = data.transaction.units
+                    };
+
+                    // flip the quantity on sell actions
+                    if (qcOrder.Direction == OrderDirection.Sell)
+                    {
+                        fill.FillQuantity *= -1;
+                    }
+                    OnOrderEvent(fill);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Obtain the active open Trade List from Oanda.
+        /// </summary>
+        /// <param name="requestParams"></param>
+        /// <returns></returns>
+        public TradesResponse GetTradeList(Dictionary<string, string> requestParams = null)
+        {
+            var requestString = EndpointResolver.ResolveEndpoint(OandaEnvironment, Server.Account) + "accounts/" + AccountId + "/trades";
+            return MakeRequest<TradesResponse>(requestString, "GET", requestParams);
+        }
+        
+        /// <summary>
+        /// Modify the specified order, updating it with the parameters provided
+        /// </summary>
+        /// <param name="orderId">the identifier of the order to update</param>
+        /// <param name="requestParams">the parameters to update (name, value pairs)</param>
+        public void UpdateOrder(long orderId, Dictionary<string, string> requestParams)
+        {
+            var orderRequest = EndpointResolver.ResolveEndpoint(OandaEnvironment, Server.Account) + "accounts/" + AccountId + "/orders/" + orderId;
+
+            var order = MakeRequest<DataType.Order>(orderRequest);
+            if (order != null && order.id > 0)
+            {
+                var requestString = EndpointResolver.ResolveEndpoint(OandaEnvironment, Server.Account) + "accounts/" + AccountId + "/orders/" + orderId;
+                try
+                {
+                    MakeRequestWithBody<DataType.Order>(requestString, "PATCH", requestParams);
+                     
+                } 
+                catch (Exception)
+                {
+                }
+                
+            } 
+            else
+            {
+
+                OnMessage(new BrokerageMessageEvent(BrokerageMessageType.Warning, "UpdateFailed", "Failed to update Oanda order id: " + orderId + "."));
```

Comment:
This would happen that when the order in question is already filled/cancelled.

I've added an OnOrderEvent declaring the order as invalid back to QC with the message 'Order currently does not exist with order id: {0}.'

Would this suffice?

If it's a 500 error or timeout then the method will throw an exception before reaching that statement.